log_drains.go

```diff
@@ -15,6 +17,15 @@ type LogDrain struct {
 	URL   string `json:"url"`
 }
 
+type errorRes struct {
+	Error string `json:"error"`
+}
+
+type LogDrainRes struct {
+	LogDrain LogDrain `json:"drain"`
```

Comment:
```suggestion
	LogDrain *LogDrain `json:"drain,omitempty"`
```

I assume `drain` is not included in the resulting JSON if there is an `errors` attribute? With this, `LogDrain` would equal `nil` if `drain` is not in the JSON. But I don't know if it applies to this usecase :)